tests/lib/rules/computed-property-spacing.js

```diff
@@ -782,7 +782,7 @@ ruleTester.run("computed-property-spacing", rule, {
         {
             code: "A = class { [ a ](){} get [ b ](){} set [ c ](foo){} static [ d ](){} static get [ e ](){} static set [ f ](bar){} }",
             output: "A = class { [a](){} get [b](){} set [c](foo){} static [d](){} static get [e](){} static set [f](bar){} }",
-            options: ["never", {}],
```

Comment:
Why did this test need to modified?

Reply:
with having `{}`, this code was valid cause it was being mutated with its default value which is set to true in [schema](https://github.com/eslint/eslint/blob/master/lib/rules/computed-property-spacing.js#L33). 

With this change, configs are not being mutated so when user passed an empty object it will not mutate the config with default one. It will only pass an empty object. 


let me know if this is the behavior you want for the linter ?

Reply:
Rule should get the mutated object, that's the purpose of default values in schema.

Reply:
but if we are passing an empty object as an option, should it mutate then too?

Reply:
Yes. In this particular rule, we have `boolean` property `enforceForClassMembers`. If the options object is present, the rule expects this property to be set to either `true` or `false`. It doesn't check for `undefined` in order to apply default value (`true`), so if user didn't specify `enforceForClassMembers` in the object, then it should be automatically set to `true`.

This change would also negatively affect non-empty objects. For example, code like this would be now `valid` in the `use-isnan` rule:

```js
{
    code: "switch(NaN) { case foo: break; }",
    options: [{ enforceForIndexOf: true }],
},
```

we didn't specify `enforceForSwitchCase`, so the rule should behave as if `enforceForSwitchCase` was set to `true` (default), meaning that this code should be `invalid`.

Reply:
I got your point,

I was thinking then what would be the difference between `[ ]` vs `[ {} ]` ?

I did tried cloning in [these](https://github.com/eslint/eslint/pull/13034#discussion_r413592713) location, it was not fixing the issue.

Reply:
> I was thinking then what would be the difference between `[ ]` vs `[ {} ]` ?

That should be the same.
 
> I did tried cloning in [these](https://github.com/eslint/eslint/pull/13034#discussion_r413592713) location, it was not fixing the issue.

I'm guessing there might be a problem with how `lodash.cloneDeep` works:

```js
var _ = require("lodash");

var nested = {};

var obj = { x: nested, y: nested };

var clone = _.cloneDeep(obj);

console.log(clone.x === clone.y); // true
```

Maybe we should somehow clone config per rule?

Reply:
> ```js
> var _ = require("lodash");
> 
> var nested = {};
> 
> var obj = { x: nested, y: nested };
> 
> var clone = _.cloneDeep(obj);
> 
> console.log(clone.x === clone.y); // true
> ```

ummm.. 🤔 . maybe I should try `JSON.parse(JSON.stringify(obj))` instead. I will check this in the mentioned place. 


> Maybe we should somehow clone config per rule?

[here ?](https://github.com/eslint/eslint/blob/master/lib/shared/config-validator.js#L121)

Reply:
> ummm.. 🤔 . maybe I should try `JSON.parse(JSON.stringify(obj))` instead. I will check this in the mentioned place.

it worked I guess. Thanks, @mdjermanovic for pointing the lodash issue